### PR TITLE
Add invalid update registration payload to conformance tests

### PR DESCRIPTION
**Pull request by dnadales**
Part of https://github.com/input-output-hk/cardano-ledger-specs/issues/771

### CODE REVIEW

File: cardano-ledger/test/Test/Cardano/Chain/Block/Model.hs [hunk 1]
```diff
@@ -196,15 +200,21 @@ ts_prop_invalidDelegationSignalsAreRejected =
...
+            footnote $ "Signal: " ++ show block
             footnote $ "Abstract result: " ++ show abstractResult
             footnote $ "Concrete result: " ++ show concreteResult
             failure
   where
     failureProfile :: [(Int, SignalGenerator CHAIN)]
-    failureProfile = [(1, invalidDelegationGen)]
+    failureProfile = [ (1, invalidDelegationGen)
+                     , (1, invalidUpdateProposalGen)
+                     -- TODO: coming soon
```
Comment (ruhatch) [inline cardano-ledger/test/Test/Cardano/Chain/Block/Model.hs:215]: > Coming soon to an invalid generator near you

**Review by ruhatch — APPROVED** [review]
LGTM :)

**Comment by mhuesch** [comment]
@dnadales It's Monday morning and I am already being a pest 🐞

I ran `cabal new-build` with latest of this repo and found the error:

```
$ cabal new-build cardano-ledger
...
[ 8 of 50] Compiling Test.Cardano.Chain.Elaboration.Update ( test/Test/Cardano/Chain/Elaboration/Update.hs, /Users/mhueschen/Projects/IOHK-work/cardano-ledger/dist-newstyle/build/x86_64-osx/ghc-8.6.5/cardano-ledger-0.1.0.0/t/cardano-ledger-test/build/cardano-ledger-test/cardano-ledger-test-tmp/Test/Cardano/Chain/Elaboration/Update.o )

test/Test/Cardano/Chain/Elaboration/Update.hs:31:1: error:
    Could not find module ‘Ledger.Core.Omniscient’
    Use -v to see a list of the files searched for.
   |
31 | import Ledger.Core.Omniscient (signatureVKey, signatureData)
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

`stack build` works, as predicted by your changes in this PR.

I am wondering, though, whether it makes sense to have git-hash version-pins in both `stack.yaml` and `cabal.project`. It seems very easy for them to get out of sync...

Git submodules would solve this problem, I think, by moving the version-pinning function into `git` and out of the build-systems' (cabal & stack) config. I know that people don't like git submodules though.

Alternately we could force everyone to choose between cabal and stack?

**Comment by dnadales** [comment]
> @dnadales It's Monday morning and I am already being a pest ￼

Hehe, no problem if it helps us improve things :D 

I'm fine with whatever we decide. If we use `cabal` and drop `stack` we need to change our CI infrastructure, which doesn't seem like a good idea. 

If you have the energy and the time feel free to discuss this with devops :pray:

**Pull request merged**